modules/howtos/examples/transactions-example.php

```diff
@@ -0,0 +1,489 @@
+<?php
+
+require_once 'Couchbase/autoload.php';
+
+use \Couchbase\ClusterOptions;
+use \Couchbase\Cluster;
+use \Couchbase\TransactionsConfiguration;
+use Couchbase\TransactionAttemptContext;
+use Couchbase\TransactionQueryOptions;
+use \Couchbase\DurabilityLevel;
+use \Couchbase\QueryOptions;
+use \Couchbase\QueryScanConsistency;
+use \Couchbase\MutationState;
+
+$CB_USER = getenv('CB_USER') ?: 'Administrator';
+$CB_PASS = getenv('CB_PASS') ?: 'password';
+$CB_HOST = getenv('CB_HOST') ?: 'couchbase://db';
+$durabilityLevel = Couchbase\DurabilityLevel::NONE; // TODO
+
+// tag::config[]
+$options = new ClusterOptions();
+$options->credentials($CB_USER, $CB_PASS);
+
+$transactions_configuration = new TransactionsConfiguration();
+$transactions_configuration->durabilityLevel($durabilityLevel); // Couchbase\DurabilityLevel::PERSIST_TO_MAJORITY etc.
+$options->transactionsOptions($transactions_configuration);
+
+$cluster = new Cluster($CB_HOST, $options);
+// end::config[]
+
+// tag::bucket[]
+// get a reference to our bucket
+$bucket = $cluster->bucket('travel-sample');
+// end::bucket[]
+
+// tag::collection[]
+// get a reference to a collection
+$collection = $bucket->scope('inventory')->collection('airline');
+// end::collection[]
+  
+// tag::default-collection[]
+// get a reference to the default collection, required for older Couchbase server versions
+$collection_default = $bucket->defaultCollection();
+// end::default-collection[]
+
+function removeOrWarn($collection, $doc) {
+  try {
+    $collection->remove($doc);
+  }
+  catch (\Couchbase\Exception\DocumentNotFoundException $e) {
+    echo "Document does not exist.\n";
+  }
+}
+
+$testDoc = 'foo';
+removeOrWarn($collection, $testDoc);
+removeOrWarn($collection, 'doc-c');
+removeOrWarn($collection, 'docId');
+$collection->upsert('doc-a', []);
+$collection->upsert('doc-b', []);
+
+
+try {
+  $cluster->transactions()->run(
+    function (TransactionAttemptContext $ctx) use ($collection, $testDoc) {
+      $ctx->insert($collection, $testDoc, "hello");
+    }
+  );
+}
+catch (\Exception $e) {
```

Comment:
It'd be best to show best-practices error handling, same as the Java examples, e.g. catching TransactionCommitAmbiguousException and TransactionFailedException.

Reply:
OK, but leaving pending clarification of Exception classes requested above.